### ISSUE TITLE
sgx feature includes libc deps

### ISSUE THREAD

**Issue by jethrogb** [fortanix/rust-mbedtls#68]
Currently, the `sgx` Cargo feature requires the `std` Cargo feature, which in turn requires the sys `std` Cargo feature, which then sets the following defines:

* MBEDTLS_NET_C
* MBEDTLS_FS_IO
* MBEDTLS_NO_PLATFORM_ENTROPY
* MBEDTLS_DEBUG_C
* MBEDTLS_ENTROPY_C

This seems wrong, as it generates undefined symbols to libc:

```
$ objdump -t libmbedcrypto.a |grep fopen
0000000000000000         *UND*	0000000000000000 fopen
0000000000000000         *UND*	0000000000000000 fopen
0000000000000000         *UND*	0000000000000000 fopen
0000000000000000         *UND*	0000000000000000 fopen
0000000000000000         *UND*	0000000000000000 fopen
0000000000000000         *UND*	0000000000000000 fopen
0000000000000000         *UND*	0000000000000000 fopen
$ objdump -t libmbedtls.a |grep recvfrom
0000000000000000         *UND*	0000000000000000 recvfrom
```

`MBEDTLS_DEBUG_C` might be ok? Besides, the setting of the `std` feature is correct for `mbedtls`, as it influences conditional compilation.

**Comment by jack-fortanix** [comment]
I understand the issue but it is not obvious to me what resolution is preferred.

Should we add additional features to "mbedtls-sys" - "network", "platform_entropy", "fs" - and use those to signify support for those operations instead of using "std"? Leaving "std" just for things like deciding if we need to import core or not.

**Comment by jethrogb** [comment]
I think it's good to have a `std` feature in `mbedtls`. Maybe we should keep it in `-sys` as well, but then we should have the build script control what that means in terms of MbedTLS defines based on the compilation target.

**Comment by jethrogb** [comment]
Fixed in #135
